### PR TITLE
Add Alpine variant

### PR DESCRIPTION
**Pull request by J0WI**
This adds a php-fpm-alpine variant with the same dependencies installed as the Debian variants, but without Cron (#253). The `entrypoint.sh` is not fully POSIX compatible, so there are some extra dependencies for it.

### CODE REVIEW

**Comment by J0WI** [comment]
Only 64-bit works so far. @tianon do you know how I can build APCu on 32-bit?

File: Dockerfile-alpine.template [hunk 1]
```diff
@@ -0,0 +1,111 @@
+FROM php:7.1-%%VARIANT%%
+
+# entrypoint.sh dependencies
+RUN apk add --no-cache \
+    bash \
+    coreutils \
+    rsync
+
```
Comment (tilosp) [inline Dockerfile-alpine.template]: I think you should merge the top 3 RUN commands into a single one to keep it a similar as possible to the debian one.

**Comment by tilosp** [comment]
we should also update `generate-stackbrew-library.sh` similar to docker-library/wordpress#223 because debian supports different architectures than alpine.

**Comment by tianon** [comment]
```
/usr/local/include/php/ext/pcre/php_pcre.h:29:18: fatal error: pcre.h: No such file or directory
```

Adding `pcre-dev` to `.build-deps` appears to get past that error on `i386`.

**Comment by tianon** [comment]
(https://ram.tianon.xyz/post/2017/12/26/dockerize-compiled-software.html is a writeup about how I usually solve this type of issue generally when Dockerizing software)

**Comment by J0WI** [comment]
_trigger Travis again_

**Comment by J0WI** [comment]
Travis is not building for some reason.

@tianon thanks for the hint. But I'm still confused why this is working on `amd64` without `pcre-dev`.

> we should also update generate-stackbrew-library.sh similar to docker-library/wordpress#223 because debian supports different architectures than alpine.

I think this is not related to this PR. I'd like to only add Alpine images here.

File: Dockerfile-alpine.template (hunk 1 above)
Comment (J0WI) [reply Dockerfile-alpine.template]: This way is better for build caching. See also other images in the docker-library.
I can adjust the Debian images if required.

**Comment by J0WI** [comment]
Travis is done now. Only fetching from keyserver failed once, which is acceptable.

**Comment by tilosp** [comment]
> I think this is not related to this PR. I'd like to add only Alpine images here.

It is related to this pr. If we merge this now the Jenkins server will try to build the alpine variant on arm32v5 and arm32v7 which will fail because there is no base image.

**Comment by J0WI** [comment]
Okay, I can do this. But the bashbrew dependency makes it a bit tricky. I didn't found a proper way to add bashbrew to Travis, but I don't want to hardcode the architectures again.

**Comment by tianon** [comment]
FWIW, up-to-date pre-built binaries of `bashbrew` can be found at https://doi-janky.infosiftr.net/job/bashbrew/lastSuccessfulBuild/artifact/bin/

**Comment by J0WI** [comment]
@tilosp can you help me out with the bashbrew dependency on Travis?

**Comment by J0WI** [comment]
@tilosp thanks for fixing it. I would like to request a review again.

**Comment by tilosp** [comment]
@J0WI i added the `cron.sh`. This adds only 62 Bytes.

**Comment by tilosp** [comment]
@tianon please review

**Comment by tilosp** [comment]
Thanks a lot for your work!

**Pull request merged**